src/main/java/com/google/sps/servlets/AggregationServlet.java

```diff
@@ -66,19 +67,52 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     // Some mock devices so we have data in the interim
     devices.add(new ChromeOSDevice("assetId", "location", "user", "deviceId", "serialNumber"));
     devices.add(new ChromeOSDevice("12345", "California", "Jane", "ae25f1-91ce6a", "SN54321"));
+    devices.add(new ChromeOSDevice("12345", "New Jersey", "Jane", "<deviceId>", "SN12345"));
 
     return devices;
   }
 
-  public Map<String, Integer> processData(List<ChromeOSDevice> devices, AnnotatedField field) {
-    Map<String, Integer> counts = new HashMap<>();
+  public static MultiKeyMap<String, Integer> processData(List<ChromeOSDevice> devices, Set<AnnotatedField> fields) {
+    MultiKeyMap<String, Integer> counts = new MultiKeyMap<>();
 
     for (ChromeOSDevice device : devices) {
-      String fieldValue = field.getField(device);
-      Integer newVal = counts.getOrDefault(fieldValue, new Integer(0)).intValue() + 1;
-      counts.put(fieldValue, newVal);
+      String keyParts[] = new String[fields.size()];
+      Iterator<AnnotatedField> it = fields.iterator();
+      int i = 0;
+
+      while (it.hasNext()) {
+        keyParts[i++] = it.next().getField(device);
+      }
+
+      MultiKey key = new MultiKey(keyParts);
+      Integer newVal = counts.getOrDefault(key, new Integer(0)).intValue() + 1;
+
+      counts.put(key, newVal);
     }
 
     return counts;
   }
+
+  /** Used for convenience in tests when only aggregating by one field*/
+  public static MultiKeyMap<String, Integer> processData(List<ChromeOSDevice> devices, AnnotatedField field) {
```

Comment:
If this function is only used for testing, it would be better to move it to AggregationServletTest.java, and make it a helper function similar to setNewResponseWriter().